src/Patterns/PatternsHelper.php

```diff
@@ -74,6 +74,52 @@ public static function get_image_url( array $images, int $index, string $default
 
 		return $image;
 	}
+	/**
+	 * Returns the post that has the generated data by the AI for the patterns.
+	 *
+	 * @return WP_Post|null
+	 */
+	public static function get_patterns_ai_data_post() {
+		$arg = array(
+			'post_type'      => 'patterns_ai_data',
+			'posts_per_page' => 1,
+			'no_found_rows'  => true,
+			'cache_results'  => true,
+		);
+
+		$query = new \WP_Query( $arg );
+
+		$posts = $query->get_posts();
+		return isset( $posts[0] ) ? $posts[0] : null;
+	}
+
+	/**
+	 * Upsert the patterns AI data.
+	 *
+	 * @param array $patterns_dictionary The patterns dictionary.
+	 *
+	 * @return WP_Error|null
+	 */
+	public static function upsert_patterns_ai_data_post( $patterns_dictionary ) {
+		$patterns_ai_data_post = self::get_patterns_ai_data_post();
+
+		try {
+			if ( isset( $patterns_ai_data_post ) ) {
+				$patterns_ai_data_post->post_content = wp_json_encode( $patterns_dictionary );
+				wp_update_post( $patterns_ai_data_post );
+			} else {
+				$patterns_ai_data_post = array(
+					'post_title'   => 'Patterns AI Data',
+					'post_content' => wp_json_encode( $patterns_dictionary ),
+					'post_status'  => 'publish',
+					'post_type'    => 'patterns_ai_data',
+				);
+				wp_insert_post( $patterns_ai_data_post );
+			}
+		} catch ( \WP_Error $e ) {
```

Comment:
If I'm understanding the [docs](https://developer.wordpress.org/reference/functions/wp_insert_post/) correctly, the `wp_insert_post` and the `wp_update_post` returns 0 or `WP_Error` on failure, but it does not throw the error, right?

Reply:
Oh, nice catch!

I updated the code [100070a](https://github.com/woocommerce/woocommerce-blocks/pull/11659/commits/100070a8411c4fb66ba00f6c8ca61801c2c48da6). What do you think?

Reply:
I think you need to pass `true` as the second parameter to make it return `WP_Error` otherwise it will be `0` (or check if it's `0` on the if, in case we don't need the error itself).
<img width="755" alt="Screenshot 2023-11-08 at 11 44 17" src="https://github.com/woocommerce/woocommerce-blocks/assets/186112/fa43fd7e-aef3-4c54-8a35-97f6507c917c">
<img width="411" alt="Screenshot 2023-11-08 at 11 44 14" src="https://github.com/woocommerce/woocommerce-blocks/assets/186112/2eac131e-549a-40b9-aaf2-a41406fb2ad0">


Sorry I should have specified that in my prev comment 🙏